packages/manager/modules/pci/src/projects/project/storages/databases/database.service.js

```diff
@@ -0,0 +1,297 @@
+  }
+
+  editRestrictedIp(projectId, engine, databaseId, ipBlock, description) {
+    return this.$http
+      .put(
+        `/cloud/project/${projectId}/database/${engine}/${databaseId}/ipRestriction/${encodeURIComponent(
+          ipBlock,
+        )}`,
+        { description },
+      )
+      .then(({ data }) => data);
+  }
+
+  deleteBackup(projectId, engine, databaseId, backupId) {
+    return this.$http.delete(
+      `/cloud/project/${projectId}/database/${engine}/${databaseId}/backup/${backupId}`,
+    );
+  }
+
+  restoreBackup(projectId, engine, databaseId, backupId) {
+    return this.$http
+      .post(
+        `/cloud/project/${projectId}/database/${engine}/${databaseId}/backup/${backupId}/restore`,
+      )
+      .then(({ data }) => data);
+  }
+
+  getBackups(projectId, engine, databaseId) {
+    return this.$http
+      .get(
+        `/cloud/project/${projectId}/database/${engine}/${databaseId}/backup`,
+        {
+          headers: {
+            'X-Pagination-Mode': 'CachedObjectList-Pages',
+          },
+        },
+      )
+      .then(({ data }) => map(data, (backup) => new Backup(backup)));
+  }
+
+  getAvailability(projectId) {
+    return this.$http
+      .get(`/cloud/project/${projectId}/database/availability`)
+      .then(({ data }) => data);
+  }
+
+  getCapabilities(projectId) {
+    return this.$http
+      .get(`/cloud/project/${projectId}/database/capabilities`)
+      .then(({ data }) => data);
+  }
+
+  getEngines(projectId) {
+    return this.$q
+      .all({
+        availability: this.getAvailability(projectId),
+        capabilities: this.getCapabilities(projectId),
+        prices: this.CucPriceHelper.getPrices(projectId),
+      })
+      .then(({ availability, capabilities, prices }) => {
+        availability.forEach((plan) => {
+          set(
+            plan,
+            'hourlyPrice',
+            get(
+              prices,
+              `databases.${plan.engine}-${plan.plan}-${plan.flavor}-launch.hour.consumption`,
+              {},
+            ),
+          );
+          set(
+            plan,
+            'monthlyPrice',
+            get(
+              prices,
+              `databases.${plan.engine}-${plan.plan}-${plan.flavor}-launch.month.consumption`,
+              {},
+            ),
+          );
+          set(
+            plan,
+            'flavor',
+            find(capabilities.flavors, { name: plan.flavor }),
+          );
+          set(plan, 'plan', find(capabilities.plans, { name: plan.plan }));
+        });
+        return capabilities.engines.map(
+          (engine) =>
+            new Engine(
+              engine,
+              availability,
+              capabilities.plans,
+              capabilities.flavors,
+            ),
+        );
+      });
+  }
+
+  getDatabases(projectId, engine) {
+    return this.$http
+      .get(`/cloud/project/${projectId}/database/${engine}`, {
+        headers: {
+          'X-Pagination-Mode': 'CachedObjectList-Pages',
```

Comment:
No need to extend pagination? By default only 10 will be fetched

Reply:
Was not aware of that! Thanks!
Added a size now :)